CMakeLists.txt

```diff
@@ -47,6 +47,16 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Change CMake default O3 to O2.
+string(REGEX REPLACE "([\\/\\-]O)3" "\\12"
+    CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+string(REGEX REPLACE "([\\/\\-]O)3" "\\12"
+    CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+string(REGEX REPLACE "([\\/\\-]O)3" "\\12"
+    CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_MINSIZEREL}")
+string(REGEX REPLACE "([\\/\\-]O)3" "\\12"
+    CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
+
```

Comment:
No. These variables are for users to set. This makes it _impossible_ for users to compile with `-O3`.

Reply:
If you want to compile with different flags, set `-DCMAKE_CXX_FLAGS_RELEASE="-O2"` at the command line.

Reply:
How do we change the default?

Reply:
There's a new feature we can optionally support in CMake 3.19 that will allow us to ship recommended defaults for these flags. But they absolutely do not belong _in_ the CMakeLists.txt.

https://cmake.org/cmake/help/v3.19/manual/cmake-presets.7.html

Reply:
I opened a PR (#5506) that uses `-O2` in the Release preset for GCC-compatible compilers. This regetx will never match for MSVC (there's no `/O3`) so I left that untouched.